### PR TITLE
Use support-library to handle the permissions for the `ShadowActivity`.

### PR DESCRIPTION
**Pull request by boyw165**

### CODE REVIEW

File: lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java [hunk 1]
```diff
@@ -23,12 +28,32 @@ protected void onNewIntent(Intent intent) {
...
+            ActivityCompat.requestPermissions(this,
+                                              yetGrantedPerms.toArray(new String[permNum]),
+                                              42);
+        }
     }
 
     @Override
-    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String[] permissions,
```
Comment (boyw165) [inline lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java]: Add `@NonNull` annotation.

File: lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
 
-@TargetApi(Build.VERSION_CODES.M)
-public class ShadowActivity extends Activity {
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class ShadowActivity extends AppCompatActivity {
```
Comment (tbruyelle) [inline lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java]: What is the benefit for a non-ui activity to subclass `AppCompatActivity` ? I don't find any...

File: lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java [hunk 3]
```diff
@@ -23,12 +28,32 @@ protected void onNewIntent(Intent intent) {
...
+            }
+        }
+
+        // Request permissions if any.
+        int permNum = yetGrantedPerms.size();
+        if (permNum > 0) {
+            ActivityCompat.requestPermissions(this,
+                                              yetGrantedPerms.toArray(new String[permNum]),
+                                              42);
+        }
```
Comment (tbruyelle) [inline lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java]: What is the goal of this ? Your PR has no comments.

Comment (tarasantoshchuk) [inline lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java]: We perform similar checks in RxPermissions class, before starting ShadowActivity.
I believe that we don't need to recheck permissions state here.

**Pull request closed without merging**